hagelslag/data/ZarrModelGrid.py

```diff
@@ -58,8 +58,10 @@ def load_data(self):
         f = s3fs.S3Map(root=path, s3=fs, check=False)
         files.append(f)
 
-        ds = xr.open_mfdataset(files, engine='zarr').load()
-        array = ds[self.variable].values.astype('float32')
+        ds = xr.open_mfdataset(files, engine='zarr', parallel=True).load()
+        arr = ds[self.variable].values.astype('float32')
+        dummy_forecast_hour_00 = np.zeros((1, arr.shape[1], arr.shape[2]))
```

Comment:
First, you should only need to create this field if forecast hour 0 is included in the range of forecast hours requested. Second, could you load the appropriate HRRR anl file here rather than doing the dummy forecast hour 0? In the slack I was a bit ambiguous about what the anl file means, but in this case it does make sense to pull the anl file and concatenate when forecast hour 0 is included in the request.

Reply:
Yes, I can use the analysis file in place of the dummy, but it could be complicated by different variable names between the two - such as `HAIL_1hr_max_fcst` vs `HAIL_max_fcst` (where the '1hr' only seem to present in the forecast but not analysis files).

As for the zero hour forecast, I think it is needed for the extract_attributes() function which relies on having the previous forecast hour than is given in the configuration.

Reply:
If the `1hr` is the only consistent difference between the variable names in the two files, could you just do `variable_name.replace("_1hr", "")` when accessing the variable from the anl file? extract_model_attributes modifies the range of forecast hours but does at the input to ModelOutput rather than by modifying attributes of an existing class. You should only have to worry about loading the anl file if forecast hour 0 is explicitly included in the array of `self.forecast_hours`.

Reply:
Yep, that makes sense.